specification/resources/resource-manager/Microsoft.Authorization/stable/2024-04-01/policyAssignments.json

```diff
@@ -0,0 +1,1050 @@
+        },
+        "policyDefinitionId": {
+          "type": "string",
+          "description": "The ID of the policy definition or policy set definition being assigned."
+        },
+        "scope": {
+          "type": "string",
+          "description": "The scope for the policy assignment.",
+          "readOnly": true
+        },
+        "notScopes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The policy's excluded scopes."
+        },
+        "parameters": {
+          "description": "The parameter values for the assigned policy rule. The keys are the parameter names.",
+          "$ref": "#/definitions/ParameterValues"
+        },
+        "description": {
+          "type": "string",
+          "description": "This message will be part of response in case of policy violation."
+        },
+        "metadata": {
+          "type": "object",
+          "description": "The policy assignment metadata. Metadata is an open ended object and is typically a collection of key value pairs."
+        },
+        "enforcementMode": {
+          "type": "string",
+          "description": "The policy assignment enforcement mode. Possible values are Default and DoNotEnforce.",
+          "enum": [
+            "Default",
+            "DoNotEnforce"
+          ],
+          "x-ms-enum": {
+            "name": "enforcementMode",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Default",
+                "description": "The policy effect is enforced during resource creation or update."
+              },
+              {
+                "value": "DoNotEnforce",
+                "description": "The policy effect is not enforced during resource creation or update."
+              }
+            ]
+          },
+          "default": "Default"
+        },
+        "nonComplianceMessages": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NonComplianceMessage"
+          },
+          "x-ms-identifiers": [
+            "message",
+            "policyDefinitionReferenceId"
+          ],
+          "description": "The messages that describe why a resource is non-compliant with the policy."
+        },
+        "resourceSelectors": {
+          "type": "array",
+          "items": {
+            "$ref": "../../common/v1/types.json#/definitions/ResourceSelector"
+          },
+          "x-ms-identifiers": [],
+          "description": "The resource selector list to filter policies by resource properties."
+        },
+        "overrides": {
+          "type": "array",
+          "items": {
+            "$ref": "../../common/v1/types.json#/definitions/Override"
+          },
+          "x-ms-identifiers": [],
+          "description": "The policy property value override."
+        },
+        "isSystemPolicy": {
+          "type": "boolean",
```

Comment:
consider making it an extensible enum instead of a boolean.

Reply:
In previous version, policy assignments were only created by end customers. With this new API version, we're aiming to introduce a feature where policy assignments can be created on a customer scope by Microsoft, hence the system naming. We don't see a case where an additional actor creates policy assignment, and think a boolean would suffice. Please let me know if there are additional concerns.